ui/v2.5/src/components/Settings/Tasks/DirectorySelectionDialog.tsx

```diff
@@ -15,12 +16,29 @@ interface IDirectorySelectionDialogProps {
   animation?: boolean;
   initialPaths?: string[];
   allowEmpty?: boolean;
+  allowPathSelection?: boolean;
+  message?: string;
+  header?: string;
+  icon?: IconDefinition;
+  acceptButtonText?: string;
+  acceptButtonVariant?: "danger" | "primary" | "secondary";
   onClose: (paths?: string[]) => void;
 }
 
 export const DirectorySelectionDialog: React.FC<
   IDirectorySelectionDialogProps
-> = ({ animation, allowEmpty = false, initialPaths = [], onClose }) => {
+> = ({
+  animation,
+  allowEmpty = false,
+  initialPaths = [],
+  allowPathSelection = true,
```

Comment:
This entire concept is a bit broken from a design perspective. The component is a `DirectorySelectionDialog`, but we have a flag to disable the entire purpose of the dialog? I think there's a better way to do this.